test/publisher/index.ts

```diff
@@ -180,56 +184,57 @@ describe('Publisher', () => {
-      // to gain access to the private field after publisher init
-      tracingPublisher['tracing'] = undefined;
-    });
-    it('should not instantiate a tracer when tracing is disabled', () => {
-      tracingPublisher = new Publisher(topic);
-      assert.strictEqual(tracingPublisher['tracing'], undefined);
-    });
-
-    it('should instantiate a tracer when tracing is enabled through constructor', () => {
-      tracingPublisher = new Publisher(topic, enableTracing);
-      assert.ok(tracingPublisher['tracing']);
-    });
-
-    it('should instantiate a tracer when tracing is enabled through setOptions', () => {
-      tracingPublisher = new Publisher(topic);
-      tracingPublisher.setOptions(enableTracing);
-      assert.ok(tracingPublisher['tracing']);
-    });
-
-    it('should disable tracing when tracing is disabled through setOptions', () => {
-      tracingPublisher = new Publisher(topic, enableTracing);
-      tracingPublisher.setOptions(disableTracing);
-      assert.strictEqual(tracingPublisher['tracing'], undefined);
+      exporter.reset();
     });
 
     it('export created spans', () => {
-      tracingPublisher = new Publisher(topic, enableTracing);
-
       // Setup trace exporting
-      const provider: BasicTracerProvider = new BasicTracerProvider();
-      const exporter: InMemorySpanExporter = new InMemorySpanExporter();
-      provider.addSpanProcessor(new SimpleSpanProcessor(exporter));
-      provider.register();
-      opentelemetry.trace.setGlobalTracerProvider(provider);
+      tracingPublisher = new Publisher(topic, enableTracing);
 
       tracingPublisher.publish(buffer);
-      assert.ok(exporter.getFinishedSpans());
+      const spans = exporter.getFinishedSpans();
+      assert.notStrictEqual(spans.length, 0, 'has span');
+      const createdSpan = spans.concat().pop()!;
+      assert.strictEqual(
+        createdSpan.status.code,
+        opentelemetry.SpanStatusCode.UNSET
+      );
+      assert.strictEqual(
+        createdSpan.attributes[GeneralAttribute.NET_PEER_NAME],
+        'PROJECT_ID'
+      );
+      assert.strictEqual(
+        createdSpan.attributes[MessagingAttribute.MESSAGING_OPERATION],
+        ''
+      );
+      assert.strictEqual(
+        createdSpan.attributes[MessagingAttribute.MESSAGING_SYSTEM],
+        'pubsub'
+      );
+      assert.strictEqual(
+        createdSpan.attributes[MessagingAttribute.MESSAGING_DESTINATION],
+        topic.name
+      );
+      assert.strictEqual(
+        createdSpan.attributes[MessagingAttribute.MESSAGING_DESTINATION_KIND],
+        'topic'
+      );
+      assert.strictEqual(createdSpan.name, 'topic-name send');
+      assert.strictEqual(
+        createdSpan.kind,
+        SpanKind.PRODUCER,
+        'span kind should be PRODUCER'
+      );
+      assert.ok(spans);
     });
   });
 
   describe('publishMessage', () => {
     const data = Buffer.from('hello, world!');
-    const spy = sandbox.spy();
+    // const spy = sandbox.spy();
```

Comment:
Double checking - should this line go away? I think the answer is yes, but I want to make sure.

Reply:
Yeah, there is now a `spy` defined at the top of the unit test block. Only way I was able to get things work and after four days I started to give up 🙈